app/src/main/java/cloudcity/LoggingServiceExtensions.java

```diff
@@ -169,11 +165,35 @@ private static NetworkDataModel getCloudCityData() {
         dataModel.setSpeed(location.getSpeed() * 3.6);
 
         dataModel.setCellData(getCellInfoModel(category, currentCell));
-        dataModel.setValues(getMeasurementsModel(category, currentCell));
+        // Lets initialize our MeasurementModel for sending from the registered cell model, then overwrite it's values
+        // with what we found in the SignalInformation
+        MeasurementsModel modelForSending = getMeasurementsModel(category, currentCell);
+        updateMeasurementModelByCell(modelForSending, currentSignal);
+
+        dataModel.setValues(modelForSending);
 
         return dataModel;
     }
 
+    /**
+     * Finds a registered cell information in a list of cell informations
+     * @param cellList
+     * @return the first registered cell in the list, or null if no registered cells were found
+     */
+    public static CellInformation findRegisteredCell(@NonNull List<CellInformation> cellList) {
+        CellInformation retVal = null;
+
+        for (CellInformation ci: cellList) {
+            if (!ci.isRegistered()) {
+                continue;
+            }
+
+            retVal = ci;
+        }
+
+        return retVal;
+    }
```

Comment:
correct me if i'm wrong but nowhere does the javadoc mention "first" or "last". the method just says "finds a registered cell from the list of cells". It gives zero guarantee about it being the first or the last one - while yes, in this particular implementation case it will be the last one, however there's also a possibility that only one cell can be registered at a time.

we're just interested in "whichever is registered" here.

Reply:
resolving.